### ISSUE TITLE
Load modify and save file fails on network addresses

### ISSUE THREAD

**Issue by CoolDixon** [rabanti-github/NanoXLSX#31]
If I try to open an already existing Excel file, update it and then save I get this exception:
![Screenshot 2022-09-22 at 06 37 52](https://user-images.githubusercontent.com/5981332/191659571-98c45ccf-f6ae-49b1-8f96-d8412a2a1aa2.png)

So the obvious move it to set the filename (which should have been set automatically on load), but then it fails to save and also corrupts the file!
![Screenshot 2022-09-22 at 06 37 22](https://user-images.githubusercontent.com/5981332/191659611-1f6f7daf-2cc2-4d7b-8a5f-6940bcb77902.png)
It works fine on a "C:\path" kind of path, but fails on network "\\server\path" addresses. But the strange thing is that it do hit the file as it manage to corrupt it
![Screenshot 2022-09-22 at 06 36 52](https://user-images.githubusercontent.com/5981332/191659641-28e06931-a88b-4918-9865-464c73665650.png)

I'm not including a test file here as it happens with any file

**Comment by rabanti-github** [comment]
Hi,
Thanks for the additional report. I will have a look into this too.
To be honest, I never tested it on UNC paths, because I thought this is handled by .NET IO. I will check this and provide a fix.

About setting the file name when loading: I am not sure whether this should be the default behavior, because overriding the existing file may lead to lost data, like diagrams, printer settings, pivots etc.
Maybe I add the possibility as ImportOption. Anyway, I will think about it and hopefully provide an accurate solution.

**Comment by CoolDixon** [comment]
Overriding will lose pivots? Can't everything be leaved as it is but the changes made? 

For more details, I have a personal finance Excel file where the golden sheet is the pivot table one that takes the data from the transactions sheet that in turns receive new data from the per-bank sheets. My app read different banks downloaded statements and create new per-bank sheets and then detects what are the new transactions from last import. My goal is to be able to open my finance Excel file, add new worksheet and delete old one (If I could insert rows at the top It would be posible to do it in other way, but is not the case) and then save it without affecting the pivot or any other sheets I did not updated in the process

**Comment by rabanti-github** [comment]
Hi,
This is unfortunately not how this library works.
It is a simple (originally) writer with reader functionality, but not an editor.
Everything that can be written can also be read... Everything else is discarded.
The main purpose of NanoXLSX (and PicoXLSX) is to create reports or machine-readable exports, whereas NanoXLSX also can read imports to gather data for further processing in the application.

If you want to edit existing Excel Files, you should better use POI / NPOI or even the slow Office interop functions.

Of course I will look into the UNC problem and the style reader problem, but sadly, I cannot provide you preserving editor functionality with this fix.

**Comment by rabanti-github** [comment]
I tried to look into the issue about the UNC path.
However, I could not reproduce it. It is probably not an issue about the path (is handled by System.IO as expected) if all necessary access rights are there. Have you checked your write permissions to this path?
Have you closed the file? Excel or the Windows preview are notorious for locking the read/write stream of Excel files.
Or have you tried to remove a worksheet in a workbook with only one worksheet in it? This should throw another error. 
And the inner exception is a little bit suspicious since from the very bottom of the creation process.
Therefore, I need more information:
- Currently used version of NanoXLSX (v2.0.1?)
- Origin of your sample file (which Office version?) --> Not necessary if a sample file is provided
- A sample file that is not yet corrupted but would fail on your system, using the sample code
- Used .NET framework and OS version
- Stacktrace, if available

Thank you in advance

**Comment by CoolDixon** [comment]
> If you want to edit existing Excel Files, you should better use POI / NPOI or even the slow Office interop functions.

This is fair enough. I started using PicoXLSX only to read, but you know how it goes when you start programming, as soon as you achieve one goal you start thinking in the more complex solution... its a never ending story, so if at the beginning it was fine to just create a XLSX file and merge with my master one by hand It turned to try to edit the master file directly and avoid my manual intervention. But I don't want to use Office interop, that's why I found Pico (and later Nano) XLSX implementation

> I need more information:

Well my setup is a little uncommon I guess. My host is MacOS running Monterrey 12.6, but my dev box is Windows 10 and I connect through RDP. That's why my Mac desktop is automatically connected to the remote PC by UNC
I use latest 2.0.1 NanoXLSX. Office 365 (Excel version 16.65 (22091101).
I use netCore 6 and Windows 10 build  

When I was preparing the sample I found out the solution and the cause of the problem. Till now I was trying to update the file so I set Workbook.Filename to the same file path as the filename I loaded (see my code in the original post). But now that I tried to send you both the original sample and corrupted files I specified different output filename and it worked fine. 

So to summarise, if you try to update the same file you load and then save it (as in my original code) it fails.

This fails:
`
string Filename = @"//tsclient/Desktop/Sample.xlsx";
Workbook wb = Workbook.Load(Filename);
wb.Filename = Filename;
wb.RemoveWorksheet("Sheet2");
wb.Save();
`

![Screenshot 2022-09-23 at 09 42 55](https://user-images.githubusercontent.com/5981332/191913467-4dc19022-3261-41aa-95b2-16d1ab7acc0c.png)

This works fine:
`
string inFilename = @"//tsclient/Desktop/Sample.xlsx";
string outFilename = @"//tsclient/Desktop/CorruptedSample.xlsx";
Workbook wb = Workbook.Load(inFilename);
wb.Filename = outFilename;
wb.RemoveWorksheet("Sheet2");
wb.Save();
`

**Comment by CoolDixon** [comment]
I tried in local filesystem and it works fine as well
`
string Filename = @"C:\_Local\Sample.xlsx";
Workbook wb = Workbook.Load(Filename);
wb.Filename = Filename;
wb.RemoveWorksheet("Sheet2");
wb.Save();
`
I would close this issue. It is a very corner case I think that don't deserve you time

 Thanks a lot anyway! ;)

**Comment by rabanti-github** [comment]
OK thanks.
To be honest, although I have experience on several platforms, I have not really tested (integrational) the library on many platforms and over network resources.
Normally, I would handle such an issue on the job as following:
1. Create the file locally in a temp folder
2. Copy or move the file to the destination folder in the network

Therefore, I will close this issue for now. However, If there are further issues with streams, I will of course have a look into it.
I hope this is OK for you.
Thank you anyway for the report
